### ISSUE TITLE
TVOS text doesn't scroll down in a ScrollView

### ISSUE THREAD

**Issue by jasmeet447** [react-native-community/react-native-tvos#82]
Hii, I've been trying to make a scrollable text component for tvos like the one we see in terms and conditions box in a lot of applications. The code works fine for android tv but the text doesn't scroll down in tvos.
REACT NATIVE VERSION: 0.61.2
STEPS TO REPRODUCE
The code snippet is as follows:
<ScrollView style={
{
flex: 1,
paddingTop: getScaledValue(50)
}}
contentContainerStyle={{
alignItems: 'center',
justifyContent: 'center'
}}
removeClippedSubviews={false}
isTVScrollable={true}
< Text style={styles.textH2} >
This is the text portion that needs to be scrolled .............
< /Text >
< /ScrollView >

Any help will be appreciated!

**Comment by dlowder-salesforce** [comment]
I'll take this up and see if I can get it to work.  This seems to be a general issue with tvOS, not just a React Native issue.  Some suggested fixes are in https://developer.apple.com/forums/thread/19309 , I'll play around with it and see what works.

**Comment by dlowder-salesforce** [comment]
@jasmeet447 here's something that will hopefully get you unblocked, and at least make the scrolling work.  I think it definitely needs some more tweaks, but it is a step in the right direction.

1. Make the following patch in the native iOS/tvOS React Native source code:

```
--- node_modules/react-native/React/Views/ScrollView/RCTScrollView.h	2020-08-18 15:08:21.000000000 -0700
+++ node_modules/react-native/React/Views/ScrollView/RCTScrollView.h	2020-08-18 15:08:21.000000000 -0700
@@ -10,11 +10,11 @@
 #import <React/RCTAutoInsetsProtocol.h>
 #import <React/RCTEventDispatcher.h>
 #import <React/RCTScrollableProtocol.h>
-#import <React/RCTView.h>
+#import <React/RCTTVView.h>
 
 @protocol UIScrollViewDelegate;
 
-@interface RCTScrollView : RCTView <UIScrollViewDelegate, RCTScrollableProtocol, RCTAutoInsetsProtocol>
+@interface RCTScrollView : RCTTVView <UIScrollViewDelegate, RCTScrollableProtocol, RCTAutoInsetsProtocol>
 
 - (instancetype)initWithEventDispatcher:(RCTEventDispatcher *)eventDispatcher NS_DESIGNATED_INITIALIZER;
```

2. For the ScrollView that wraps the text block, add the property `isTVSelectable={true}`.

**Comment by jasmeet447** [comment]
@dlowder-salesforce I tried the steps that you've mentioned above, I get an error "Exception thrown while executing UI block". 
<img width="1294" alt="Screenshot 2020-08-19 at 6 10 28 PM" src="https://user-images.githubusercontent.com/44059494/90636120-c7270500-e247-11ea-9f07-e4eb963a3581.png">
I think it doesn't recognize the RCTTVView class. Can you tell me a possible explanation for the issue?
Here is a reference that might help you get it:

**Comment by dlowder-salesforce** [comment]
The error screen you showed is what happens if you don’t recompile the app before running it.

**Comment by dlowder-salesforce** [comment]
Thanks @jasmeet447 -- I'm going to keep working on a fully native solution for this so that fewer events have to cross the bridge.  [This project](https://github.com/douglashill/KeyboardKit) aims to provide full keyboard support for iOS apps -- I'm looking at what was done for the scroll view to see if it can be applied here.

**Comment by dlowder-salesforce** [comment]
Fixed in 0.63.1-1
